use_cases/eluc/prescriptors/esp/unileaf_prescriptor.py

```diff
@@ -0,0 +1,305 @@
+        :return: True if only 1 action is defined, False otherwise
+        """
+        return len(self.cao_mapping["actions"]) == 1
+
+    @staticmethod
+    def _is_scalar(prescribed_action):
+        """
+        Checks if the prescribed action contains a single value, i.e. a scalar, or an array.
+        A prescribed action contains a single value if it has been prescribed for a single context sample
+        :param prescribed_action: a scalar or an array
+        :return: True if the prescribed action contains a scalar, False otherwise.
+        """
+        return prescribed_action.shape[0] == 1 and prescribed_action.shape[1] == 1
+
+    @staticmethod
+    def get_context_field_names(config: Dict[str, Any]) -> List[str]:
+        """
+        Returns the list of Context column names
+        :param config: the ESP experiment config dictionary
+        :return: the list of Context column names
+        """
+        nn_inputs = config["network"]["inputs"]
+        contexts = [nn_input["name"] for nn_input in nn_inputs]
+        return contexts
+
+    @staticmethod
+    def get_action_field_names(config: Dict[str, Any]) -> List[str]:
+        """
+        Returns the list of Action column names
+        :param config: the ESP experiment config dictionary
+        :return: the list of Action column names
+        """
+        nn_outputs = config["network"]["outputs"]
+        actions = [nn_output["name"] for nn_output in nn_outputs]
+        return actions
+
+    @staticmethod
+    def get_fitness_metrics(config: Dict[str, Any]) -> List[str]:
+        """
+        Returns the list of fitness metric names (Outcomes) to optimize.
+        :param config: the ESP experiment config dictionary
+        :return: the list of fitness metric names
+        """
+        metrics = config["evolution"]["fitness"]
+        fitness_metrics = [metric["metric_name"] for metric in metrics]
+        return fitness_metrics
+    
+    def prescribe_land_use(self, cand_id: str, results_dir: Path, context_df: pd.DataFrame) -> pd.DataFrame:
+        gen = int(cand_id.split('_')[0])
+        candidate_filename = results_dir / f"{gen}" / f"{cand_id}.h5"
+        candidate = load_model(candidate_filename, compile=False)
+        
+        encoded_context_df = self.data_encoder.encode_as_df(context_df)
+
+        reco_land_use = self.prescribe(candidate, encoded_context_df)
+        reco_df = pd.DataFrame(reco_land_use["reco_land_use"].tolist(), columns=constants.RECO_COLS)
+        context_actions_df = self._reco_to_context_actions(reco_df, encoded_context_df)
+
+        context_actions_df = context_actions_df.set_index(context_df.index)
+
+        return context_actions_df
+    
+    def predict_metrics(self, context_actions_df: pd.DataFrame) -> tuple:
+        eluc_df = self.predict_eluc(context_actions_df)
+        change_df = self.compute_percent_changed(context_actions_df)
+
+        return eluc_df, change_df
+        
```

Comment:
Added these 2 bottom functions to implement the Prescriptor.py abstract class. They are just simple calls of already made functions.

Reply:
Maybe we should make theme more generic in a later iteration, e.g. `prescribe(...)` and `compute_metrics(...)`.